finetuner/tuner/base.py

```diff
@@ -210,12 +220,86 @@ def _get_data_loader(
         """Get framework specific data loader from the input data."""
         ...
 
+    @staticmethod
+    def _get_num_batches(docs: 'DocumentSequence', batch_size: int) -> int:
+        """Get the number of batches from a document sequence"""
+        n = len(docs)
+        return n // batch_size + 1 * (n % batch_size)
+
     @abc.abstractmethod
     def _train(self, data: AnyDataLoader):
         """Train the model on given labeled data"""
         ...
 
-    @abc.abstractmethod
-    def _eval(self, data: AnyDataLoader):
-        """Evaluate the model on given labeled data"""
-        ...
+    def _eval(
+        self,
+        query_data: 'DocumentSequence',
+        index_data: Optional['DocumentSequence'],
+        label: str,
+        limit: int,
+        distance: str,
+        batch_size: int,
+        preprocess_fn: Optional['PreprocFnType'] = None,
+        collate_fn: Optional['CollateFnType'] = None,
+    ):
+        """Embed the evaluation data and ompute the evaluation metrics"""
+
+        self.state.num_batches_query = self._get_num_batches(query_data, batch_size)
+        self.state.num_batches_index = (
+            self._get_num_batches(index_data, batch_size) if index_data else 0
+        )
+        self.state.batch_index = 0
+
+        self._trigger_callbacks('on_val_begin')
```

Comment:
`val` or `test`? Normally in `validation` split you just control the `loss` and in `test` split you check the actual evaluation metric related to the business value

Reply:
we made the decision to bypass the validation loss completely. Now it's only two phases, train and eval. In train you train and calculate the training loss, in eval you compute the search related metrics

Reply:
why was this decision taken? validation step is a common practice.

But anyway, the naming is still confusing to much of the community

Reply:
I think it's still eval though because this step happens at the end of every epoch, not at the end of training

Reply:
To me `val` reminds me of `validation` loss not about `eval`. The naming is important. You can say `on_eval_begin` not on `val_begin`.

`Validation` tends to be about loss and `evaluation` about other metrics